library/js/oeUI/headerTitleAction.js

```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ * Expand Contract State and Show /Hide State js
+ *
+ * @package   OpenEMR
+ * @link      http://www.open-emr.org
+ * @author Ranganath Pathak <pathak@scrs1.org>
+ * @copyright Copyright (c) 2018 Ranganath Pathak <pathak@scrs1.org>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+?>
+$(document).ready(function () {
+    $('.expand_contract').click(function () {
+        var elementTitle = $(this).prop('title');
+        var expandTitle = '<?php echo xlj('Click to Contract and set to henceforth open in Centered mode'); ?>';
```

Comment:
Very cool to see you using the newer js escaping function. Note that the js escaping function add quotes at the end, so you need to remove the quotes:
```
var expandTitle = <?php echo xlj('Click to Contract and set to henceforth open in Centered mode'); ?>;
```

Reply:
will do, explain the use of` attr_url` function

Reply:
That is case in html (not js portion) code, where you encode url parameter (so doesn't break web url address) and then escape for html (to prevent cross scripting attacks). Here is an example:
https://github.com/openemr/openemr/blob/e10a69185a5843ec2c0f1616bc03b16bf6af3aa6/interface/patient_file/reminder/patient_reminders.php#L103

Reply:
When I used it on the entire href value it caused all links to fail especially in the OemrUI class

Reply:
@bradymiller Just realized you are using json_encode() to escape here. Good choice, especially for languages but, do you think maybe, in the case here, should check strings are utf8 encoding compliant before passing on? I used this in custom_reports before calling mPDF: preg_match('//u', $string). Maybe not an issue but with languages better to be safe..

Reply:
sorry, I meant in your xlj() function...

Reply:
@bradymiller 

This unfortunately breaks when the whole href is passed as a variable :(

Reply:
hi @zbig01 ,
I am assuming you are talking about the attr_url function. Note the url encoding is done via urlencode() function (http://php.net/manual/en/function.urlencode.php) which is only meant to encode the parameters in the url. If you try to encode the entire url, it will break.
-brady

Reply:
hi @sjpadgett ,
I kind of figured things would come up, which is why used the wrapper (ie. can address things as needed). What happens if I do the xlj() function on non UTF8 encoded stuff? Does it mangle characters or worse yet, is there way to get around the escaping (ie. security vulnerabilties)?
https://github.com/openemr/openemr/blob/088808c07956ca3fe7e62224cd51594dca902d41/library/htmlspecialchars.inc.php#L157-L160
(I know I could try it myself, but just being lazy :) )
-brady

Reply:
@sjpadgett ,
If we do indeed need to check for UTF8 within the function, is this a good route: http://php.net/manual/en/function.mb-check-encoding.php
-brady

Reply:
I believe utf-8 is required for strings using json_encode. Not sure what its error level and I don't recall ever seeing this as a problem in OpenEMR but, if you're anything like me with this guy Murphy (Murphy law renown) always hanging around, would opt for safer than sorry.
I'd also opt for preg_match test instead of a multi byte function because multi byte extension may be turned off/not installed. Not sure what default is here either.